openassessment/xblock/data_conversion.py

```diff
@@ -56,13 +57,42 @@ def convert_training_examples_list_to_dict(examples_list):
     ]
 
 
-def create_rubric_dict(prompt, criteria):
+def create_prompts_list(prompt_or_serialized_prompts):
+    """
+    Construct a list of prompts.
+
+    Initially a block had a single prompt which was saved as a simple string.
+    In that case a new prompt dict is constructed from it.
+
+    Args:
+        prompt_or_serialized_prompts (unicode): A string which can either
+        be a single prompt text or json for a list of prompts.
+
+    Returns:
+        list of dict
+    """
+
+    if prompt_or_serialized_prompts is None:
+        prompt_or_serialized_prompts = ''
+
+    try:
+        prompts = json.loads(prompt_or_serialized_prompts)
```

Comment:
Oh I see what's happening now. All the configuration is still in "prompt", but may be a serialized representation of multiple prompts.

Out of curiosity, why did you choose this approach versus a new "prompts" field on the XBlock, then only use "prompt" for deprecated blocks? I'd imagine it would save you some of the complexity around serialization.